### PR TITLE
Tidy GUI labels and code

### PR DESCRIPTION
**Pull request by alexdewar**
# Description

The main reason for this PR is to change GUI labels to be more generic. For example, the panel labelled DP9800 should be labelled "Temperature monitor" as the user is not necessarily using a DP9800 any more.

I also split up `temp_control.py` into separate files for the different widgets, but aside from the label changes the code is mostly the same.

Since a571e93b65ed6803af603ae60ab5c7ec798e5bc9 we don't need any trickery to keep the Update button disabled for the cold BB control, so I've got rid of that hack.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] Optimisation (non-breaking, back-end change that speeds up the code)

## Key checklist

- [x] Pre-commit hooks run successfully (`pre-commit run -a`)
- [x] All tests pass (`pytest`)
- [ ] The documentation builds without warnings (`mkdocs build -s`)
- [x] Check the GUI still works (if relevant)
- [ ] Check the code works with actual hardware (if relevant)
- [ ] Check the `pyinstaller`-built executable works (if relevant)

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [ ] Tests have been added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: finesse/gui/temp_control.py
Comment (dalonsoa) [inline finesse/gui/temp_control.py:1]: The contents of this file have just been split between the `temperature_controller_view`, the `temperature_monitor_view` and the `temperature_plot` without functional changes, right? In other words, here you're just moving code around, aren't you? I'm checking to save reviewing those large diffs...

**Review by dalonsoa — APPROVED** [review]
Assuming the answer to the question is `yes`, all LGTM!

File: finesse/gui/main_window.py [hunk 1]
```diff
@@ -72,9 +74,13 @@ def __init__(self) -> None:
...
-        dp9800: QGroupBox = DP9800Controls()
-        tc4820_hot: QGroupBox = TC4820Controls("hot", TEMPERATURE_MONITOR_HOT_BB_IDX)
-        tc4820_cold: QGroupBox = TC4820Controls("cold", TEMPERATURE_MONITOR_COLD_BB_IDX)
+        dp9800: QGroupBox = TemperatureMonitorControl()
+        tc4820_hot: QGroupBox = TemperatureControllerControl(
+            "hot", TEMPERATURE_MONITOR_HOT_BB_IDX, allow_update=True
+        )
+        tc4820_cold: QGroupBox = TemperatureControllerControl(
+            "cold", TEMPERATURE_MONITOR_COLD_BB_IDX, allow_update=False
+        )
```
[the remark this answers is not in the document]
Comment (alexdewar) [reply finesse/gui/main_window.py:83]: Good spot!

File: finesse/gui/temp_control.py
Comment (alexdewar) [reply finesse/gui/temp_control.py:1]: Yes, other than this small change: 98476d0ca4fc7faf49ff89b62e5216c7e267de2a

No need to re-review the lot.

**Pull request merged**